docs/_sources/chapters/debugging/debugging-logic-errors.rst.txt

```diff
@@ -0,0 +1,94 @@
+
+When your code runs but you don't end up with the expected results, it is important to look at the values of variables being used.
+
+Let's look at a program that has a logical bug.
+
+.. sourcecode:: js
+
+   const input = require('readline-sync');
+
+   let degreesC = input.question('Temp in degrees C: ');
+   let degreesK = degreesC + 273.15;
+
+   console.log('Degrees K:', degreesK);
+
+`Run the code at repl.it <https://repl.it/@launchcode/Degrees-C-to-K-Logic-Error>`_.
+
+This program asks the user for a temperature in degrees celsius and attempts to convert it to degrees Kelvin. Degrees Kelvin differs from degrees celsius by 273.15. So if we enter 100 (celsius) we should see a converted value of 373.15. However, running the program as-is and entering 100 gives the message:
+
+::
+
+   Temp in degrees C:  100
+   Degrees K: 100273.15
+
+This is clearly incorrect. But the program does not generate an error, so it's not immediately clear what the issue is. To figure it out, we'll use ``console.log`` to see what the values of key variables are when the program runs. 
+
+Let's first make sure that the ``degreesC`` variable looks like it should.
+
+.. sourcecode:: js
+
+   const input = require('readline-sync');
+
+   let degreesC = input.question('Temp in degrees C: ');
+   console.log(degreesC);
+   let degreesK = degreesC + 273.15;
+
+   console.log('Degrees K:', degreesK);
+
+Running this with an input of 100 gives the output:
+
+::
+
+   Temp in degrees C:  100
+   100
+   Degrees K: 100273.15
+
+The second line is the value of ``degreesC``, which appears to be correct. However, on the next line, when setting ``degreesK``, we use ``degreesC`` as a numeric value in our calculation. Let's see what the data type of ``degreesC`` is (we want it to be a number).
+
+.. sourcecode:: js
+
+   const input = require('readline-sync');
+
+   let degreesC = input.question('Temp in degrees C: ');
+   console.log(typeof degreesC);
+   let degreesK = degreesC + 273.15;
+
+   console.log('Degrees K:', degreesK);
+
+Running this with an input of 100 gives the output:
+
+::
+
+   Temp in degrees C:  100
+   string
+   Degrees K: 100273.15
+
+That's it! The variable ``degreesC`` has the value 100 that we entered, but it is a string rather than a number. So when we set ``degreesK`` with the formula ``degreesC + 273.15``, we are actually performing string concatenation instead of addition.
+
+We can fix our program by converting the user's input to the number data type.
+
+.. sourcecode:: js
+
+   const input = require('readline-sync');
+
+   let degreesC = input.question('Temp in degrees C: ');
+   degreesC = Number(degreesC);
+   let degreesK = degreesC + 273.15;
+
+   console.log('Degrees K:', degreesK);
+
+Running this with an input of 100 gives the output:
+
+::
+
+   Temp in degrees C:  100
+   Degrees K: 373.15
+
+Note that after debugging we removed all of our ``console.log`` statements. Be sure to do the same when using this debugging technique.
```

Comment:
in "``console.log`` statements" the usage of statement is incorrect. Statements apply to things like `if, for, while`

Reply:
Semantics fixed.